src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -1192,6 +1167,39 @@ void LedgerConsensusImp::accept (std::shared_ptr<SHAMap> set)
     }
 }
 
+void
+LedgerConsensusImp::applyOpenAndLocalTxs (View& accum,
+    std::shared_ptr<Ledger> const& newLCL,
+        CanonicalTXSet& retries)
+{
+    // Apply transactions from the old open ledger
+    auto const oldOL = ledgerMaster_.getCurrentLedger();
+    if (oldOL->txMap().getHash().isNonZero ())
+    {
+        WriteLog (lsDEBUG, LedgerConsensus)
+            << "Applying transactions from current open ledger";
+        applyTransactions (&oldOL->txMap(),
+            accum, newLCL, retries);
+    }
+
+    // Apply local transactions
+    for (auto it : m_localTX.getTxSet ())
+    {
+        try
+        {
+            apply (accum, *it.second, tapNONE, getConfig(),
+                deprecatedLogs().journal("LedgerConsensus"));
+        }
+        catch (...)
```

Comment:
I know you basically moved this code around, but this still is disconcerting... I think we should put an assert there (because if it shows up in debug, we want to know) and a log too (because if someone has that cleverly malformed transaction, we'd love to know...)